python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -0,0 +1,471 @@
+        """
+        return len(self._meta.input_types)
+
+    @property
+    def points(self):
+        return self._geo._points
+
+    @property
+    def multipoints(self):
+        return self._geo._multipoints
+
+    @property
+    def lines(self):
+        return self._geo._lines
+
+    @property
+    def polygons(self):
+        return self._geo._polygons
+
+    def _dump(self):
+        return (
+            f"POINTS\n"
+            f"{self.points._repr__()}\n"
+            f"MULTIPOINTS\n"
+            f"{self.multipoints._repr__()}\n"
+            f"LINES\n"
+            f"{self.lines._repr__()}\n"
+            f"POLYGONS\n"
+            f"{self.polygons._repr__()}\n"
+        )
+
+    def __repr__(self):
+        return (
+            f"GeoColumn\n"
+            f"{len(self.points)} POINTS\n"
+            f"{len(self.multipoints)} MULTIPOINTS\n"
+            f"{len(self.lines)} LINES\n"
+            f"{len(self.polygons)} POLYGONS\n"
+        )
+
+    def copy(self, deep=True):
+        """
+        Create a copy of all of the GPU-backed data structures in this
+        GeoColumn.
+        """
+        result = GeoColumn(self._geo.copy(), self._meta.copy(), self.data)
+        return result
+
+    def _copy_type_metadata(self: T, other: ColumnBase) -> ColumnBase:
+        self._data = other.data
+        return self
+
+
+class GeoColumnLocIndexer:
+    """
+    Not yet supported.
+    """
+
+    def __init__(self):
+        # Todo: Easy to implement with a join.
+        raise NotImplementedError
+
+
+class GeoColumnILocIndexer:
+    """
+    Each row of a GeoSeries is one of the six types: Point, MultiPoint,
+    LineString, MultiLineString, Polygon, or MultiPolygon.
+    """
+
+    def __init__(self, sr):
+        self._sr = sr
+
+    def __getitem__(self, index):
+        if not isinstance(index, slice):
+            mapped_index = int(self._sr.values[index])
+            return self._getitem_int(mapped_index)
+        else:
+            raise NotImplementedError
+            # This slice functionality is not ready yet
+            # return self._getitem_slice(index)
+
+    def _getitem_int(self, index):
+        type_map = {
+            "p": GpuPoint,
+            "mp": GpuMultiPoint,
+            "l": GpuLineString,
+            "ml": GpuMultiLineString,
+            "poly": GpuPolygon,
+            "mpoly": GpuMultiPolygon,
+        }
+        return type_map[self._sr._meta.input_types[index]](self._sr, index)
+
+
+class GpuGeometry:
+    def __init__(self, source, index):
+        """
+        The base class of individual GPU geometries. This and its inheriting
+        classes do not manage any GPU data directly - each GpuGeometry simply
+        stores a reference to the GeoSeries it is stored within and the index
+        of the geometry within the GeoSeries. Child GpuGeometry classes
+        contain the logic necessary to serialize and convert GPU data back to
+        Shapely.
+        """
```

Comment:
Could you add a parameters section here? I'm struggling to understand what `source` and `index` are here.